.evergreen/config.yml

```diff
@@ -1,1031 +1,82 @@
-        background: true
-        shell: bash
-        # Use different ports for KMS HTTP servers to avoid conflicts with load balancers
-        script: |-
-          set -o errexit
-          cd ${DRIVERS_TOOLS}/.evergreen/csfle
-          . ./activate-kmstlsvenv.sh
-          python -u kms_http_server.py --ca_file ../x509gen/ca.pem --cert_file ../x509gen/expired.pem --port 8100 &
-          python -u kms_http_server.py --ca_file ../x509gen/ca.pem --cert_file ../x509gen/wrong-host.pem --port 8101 &
-          python -u kms_http_server.py --ca_file ../x509gen/ca.pem --cert_file ../x509gen/server.pem --port 8102 --require_client_cert &
-          python -u kms_kmip_server.py --port 5698 &
-    - command: expansions.update
-      params:
-        updates:
-          - key: client_side_encryption_kms_tls_ca_file
-            value: ${DRIVERS_TOOLS}/.evergreen/x509gen/ca.pem
-          - key: client_side_encryption_kms_tls_certificate_key_file
-            value: ${DRIVERS_TOOLS}/.evergreen/x509gen/client.pem
-          - key: client_side_encryption_kms_endpoint_expired
-            value: 127.0.0.1:8100
-          - key: client_side_encryption_kms_endpoint_wrong_host
-            value: 127.0.0.1:8101
-          - key: client_side_encryption_kms_endpoint_require_client_cert
-            value: 127.0.0.1:8102
-          - key: client_side_encryption_kmip_endpoint
-            value: localhost:5698
-
-  "set aws temp creds":
-    - command: shell.exec
-      params:
-        shell: bash
-        script: |-
-          set -o errexit
-
-          export AWS_ACCESS_KEY_ID="${client_side_encryption_aws_access_key_id}"
-          export AWS_SECRET_ACCESS_KEY="${client_side_encryption_aws_secret_access_key}"
-          export AWS_DEFAULT_REGION="us-east-1"
-
-          pushd ${DRIVERS_TOOLS}/.evergreen/csfle
-          . ./activate-kmstlsvenv.sh
-          . ./set-temp-creds.sh
-          popd
-
-          if [ -z "$CSFLE_AWS_TEMP_ACCESS_KEY_ID" ]; then
-            echo "Failed to set AWS temporary credentials!"
-            exit 1
-          fi
+# Fail builds when pre tasks fail.
+pre_error_fails_task: true
 
-          cat <<EOT > aws-expansion.yml
-          client_side_encryption_aws_temp_access_key_id: "$CSFLE_AWS_TEMP_ACCESS_KEY_ID"
-          client_side_encryption_aws_temp_secret_access_key_key: "$CSFLE_AWS_TEMP_SECRET_ACCESS_KEY"
-          client_side_encryption_aws_temp_session_token: "$CSFLE_AWS_TEMP_SESSION_TOKEN"
-          EOT
-    - command: expansions.update
-      params:
-        file: aws-expansion.yml
+# Protect ourselves against rogue test case that runs forever. Tasks are killed after 10 minutes, which shouldn't occur
+# under normal circumstances.
+exec_timeout_secs: 600
 
-pre:
+# These pre and post rules apply to all tasks not part of a task group, which should only ever be tests against local
+# MongoDB instances. All other tasks that require special preparation should be run from within a task group
+pre: &test_setup
```

Comment:
Are `test_setup` and `test_teardown` referenced anywhere?

Reply:
They aren't; removed.